core/src/main/java/org/elasticsearch/index/query/TermsQueryParser.java

```diff
@@ -101,13 +80,14 @@ public Query parse(QueryShardContext context) throws IOException, QueryParsingEx
                     throw new QueryParsingException(parseContext, "[terms] query does not support multiple fields");
                 }
                 fieldName = currentFieldName;
-
+                
+                values = Lists.newArrayList();
```

Comment:
note that because we moved the initialization of values, we might end up passing in null even in case we are not doing a terms lookup, in case the query is invalid. One  more reason to add this bit to the validate method.